src/libs/metadata.c

```diff
@@ -214,35 +210,30 @@ static void _update(dt_lib_module_t *self, gboolean early_bark_out)
 static gboolean _postponed_update(gpointer data)
 {
   // timeout handle clearing is handled by update code
-  _update((dt_lib_module_t *)data, FALSE);
+  _update((dt_lib_module_t *)data);
 
   return FALSE;
 }
 
 static void _image_selection_changed_callback(gpointer instance, dt_lib_module_t *self)
 {
-  _update(self, FALSE);
+  _update(self);
 }
 
 static void _collection_updated_callback(gpointer instance, dt_collection_change_t query_change, gpointer imgs,
                                         int next, dt_lib_module_t *self)
 {
-  _update(self, FALSE);
-}
-
-static gboolean _draw(GtkWidget *widget, cairo_t *cr, dt_lib_module_t *self)
-{
-  if(!dt_control_running()) return FALSE;
```

Comment:
this still should IMO have this bit. it's basically saying "if darktable is closing, do nothing on this event"

Reply:
Good to do nothing if exiting dt !
But _draw() is called thousands of times and I haven't still figured out what it does in metadata on normal time...
The "draw" event is related to cairo stuff. In metadata module I don't see anything related to that. It's possible to have had some role in the past. But currently I don't see what it does. Do you have any idea ?

Reply:
Since it simply called _update with early `bark_out` (lol) then it's most likelly that it was just overzealous update mechanism. i _think_ for metadata it would be enough to just use collection/selection/mouse_over signals and it would be "enough".

if i remember correctly dt_signal_raise doesn't even raise signals if closing darktable so it avoids unnecessary handling.

Reply:
I agree with what was executed :) but I just don't figure out why the "draw" event was triggered ...

Reply:
the _draw is triggered every time widget for full iop has to be redrawn (so quite often). you could try and comment out
```c
g_signal_connect(self->widget, "draw", G_CALLBACK(_draw), self);
```
in `gui_init` and thus avoid unnecessary calls. then see what (if anything) changes, making sure that the other signals I've mentioned are handled.

Reply:
>  you could try and comment out

This is already done, isn't it ? :)